src/back-end/lib/db.ts

```diff
@@ -28,64 +32,72 @@ export async function createUser(connection: Connection, user: Omit<User, 'id' |
   return await rawUserToUser(connection, result);
 }
 
-export async function updateUser(connection: Connection, userInfo: Partial<Omit<User, 'avatarImageFile'> & { avatarImageFile: Id }>): Promise<User> {
+type UpdateUserParams = Omit<Partial<User>, 'avatarImageFile'> & { id: Id, avatarImageFile?: Id };
+
+export async function updateUser(connection: Connection, userInfo: UpdateUserParams): Promise<DatabaseValidation<User>> {
   const now = new Date();
-  const [result] = await connection('users')
+  try {
+    const [result] = await connection<RawUserToUserParams>('users')
     .where({ id: userInfo.id })
     .update({
       ...userInfo,
       updatedAt: now
-    }, ['*']);
-  if (!result) {
-    throw new Error('unable to update user');
+    } as UpdateUserParams, '*');
+    if (!result) {
+      throw new Error('unable to update user');
+    }
+    return valid(await rawUserToUser(connection, result));
+  } catch (exception) {
+    return invalid(createDatabaseError());
   }
-  return await rawUserToUser(connection, result);
 }
 
-export async function readOneUser(connection: Connection, id: Id): Promise<User | null> {
-  const result = await connection('users')
-    .where({ id })
-    .first();
-  return result ? await rawUserToUser(connection, result) : null;
+export async function readOneUser(connection: Connection, id: Id): Promise<DatabaseValidation<User | null>> {
+  try {
+    const result = await connection<RawUserToUserParams>('users')
+      .where({ id })
+      .first();
+    return valid(result ? await rawUserToUser(connection, result) : null);
+  } catch (exception) {
+    return invalid(createDatabaseError());
+  }
 }
```

Comment:
You may want to create a helper function to standardize this error handling behaviour:

```ts
type DbValidation<Valid> = Validation<Valid, null>;

type DbFn<Arg, Valid> = (connection: Connection, arg: Arg) => Promise<DatabaseValidation<Valid>>;

function tryDb<Arg, Valid>(fn: DbFn<Arg, Valid>): DbFn<Arg, Valid> { 
  return async (connection, arg) => { 
    try { 
      return await fn(connection, arg);
    } catch (e) { 
      return invalid(null);
    } 
  };
}

// Usage
export const readOneUser = tryDb<Id /*Arg*/, User /*Valid*/>(userId /*Arg*/ => async connection => { 
  const result = await connection<RawUserToUserParams>('users')
    .where({ id })
    .first();
  return valid(result ? await rawUserToUser(connection, result) : null);
});
```